### PR TITLE
Enroll flow for v2 architecture

### PR DESCRIPTION
**Pull request by diurnalist**
This is a series of pretty big changes that effectively nukes and paves the old ansible configuration/bootstrap process w/ the new enrollment flow that largely just leverages Doni.

### CODE REVIEW

**Review by Mark-Powers — APPROVED** [review]
LGTM!

File: chi_edge/cli.py [hunk 1]
```diff
@@ -11,162 +11,250 @@
...
+        )
+        print_device(device)
+
+
+@device.command(cls=BaseCommand, short_help="list registered devices")
+def list():
+    with doni_error_handler("failed to list devices"):
+        devices = doni_client().get("/v1/hardware/").json()["hardware"]
+        table = make_table()
+        table.add_column("Name")
```
Comment (super-cooper) [inline chi_edge/cli.py:161]: this is very neat :) it would be cool to add it to our other CLI tools

File: chi_edge/cli.py [hunk 2]
```diff
@@ -11,162 +11,250 @@
...
+        if application_credential_secret:
+            patch.append(
+                patch_to("application_credential_secret", application_credential_secret)
+            )
+        print_device(doni.patch(f"/v1/hardware/{uuid}/", json=patch).json())
+
+
+@device.command(cls=BaseCommand, short_help="delete registered device")
+@click.argument("device")
+@click.option("--yes-i-really-really-mean-it", is_flag=True)
```
Comment (super-cooper) [inline chi_edge/cli.py:230]: I've seen this flag before :D

File: chi_edge/cli.py [hunk 3]
```diff
@@ -178,24 +266,24 @@ def bootstrap(
...
     balena_workers = [
-        worker for worker in hardware["workers"]
-        if worker["worker_type"] == "balena"
+        worker for worker in hardware["workers"] if worker["worker_type"] == "balena"
     ]
     if not balena_workers:
-        print(f"Expected to find 1 balena worker, found {len(balena_workers)}")
-        return
+        raise click.ClickException(
+            "Device is not enrolled with support for Balena, you may need to attempt "
```
Comment (super-cooper) [inline chi_edge/cli.py:284]: Will users know what Balena is when they see this error message?

File: chi_edge/cli.py [hunk 4]
```diff
@@ -230,5 +319,79 @@ def bake(
...
+    cols = [""] + [w["worker_type"] for w in workers]
+    worker_table = make_table(*cols, header_style="blue")
+    for key in ["state", "state_details"]:
+        worker_table.add_row(*([key] + [format_value(w[key]) for w in workers]))
+    outer.add_row(worker_table)
+    console.print(Panel(outer, title=title, title_align="left"))
+
+
+@contextlib.contextmanager
+def doni_error_handler(default_message):
```
Comment (super-cooper) [inline chi_edge/cli.py:347]: this is sick

**Review by super-cooper — APPROVED** [review]
Very cool!

File: chi_edge/cli.py (hunk 3 above)
Comment (diurnalist) [reply chi_edge/cli.py:284]: We mention it in the docs, I think they will have some minimal knowledge. Practically this shouldn't happen at all, just being specific so they can attach a more detailed error message.

File: chi_edge/cli.py (hunk 1 above)
Comment (diurnalist) [reply chi_edge/cli.py:161]: Yeah, I was pretty happy w/ it overall. It doesn't pair very easily with also supporting generic output but I guess that's OK, it's pretty easy to just passthrough the JSON responses.

**Pull request merged**